src/trio/_core/_exceptions.py

```diff
@@ -63,6 +67,31 @@ class Cancelled(BaseException, metaclass=NoPublicConstructor):
     def __str__(self) -> str:
         return "Cancelled"
 
+    def __copy__(self) -> Cancelled:
+        result = Cancelled.__new__(Cancelled)
+        result.__context__ = self.__context__
+        result.__cause__ = self.__cause__
+        result.__suppress_context__ = self.__suppress_context__
+        result.__traceback__ = self.__traceback__
+        if hasattr(self, "__notes__"):
```

Comment:
The notes are part of the dict. You probably want to copy the rest of the dict as well. So maybe just `result.__dict__ = self.__dict__.copy()`?

Reply:
thanks, I threw everything out in favor of just copying `__dict__`.